### PR TITLE
Move all handler test to integration tests

### LINKED ISSUE #417: Move more tests to integration tests

### PR DESCRIPTION
**Pull request by w84thesun**
Closes #417.

### CODE REVIEW

**Review by seeforschauer — APPROVED** [review]

File: integration/commands_administration_test.go [hunk 1]
```diff
@@ -193,3 +198,132 @@ func TestCommandsAdministrationGetParameter(t *testing.T) {
 	assert.Contains(t, keys, "quiet")
 	assert.Equal(t, false, m["quiet"])
 }
+
+func TestStatisticsCommands(t *testing.T) {
+	t.Skip("TODO: response deltas investigation needed")
```
Comment (AlekSi) [inline integration/commands_administration_test.go:203]: There should be an issue URL

File: integration/commands_administration_test.go [hunk 2]
```diff
@@ -193,3 +198,132 @@ func TestCommandsAdministrationGetParameter(t *testing.T) {
...
+	ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
+
+	for name, tc := range map[string]struct {
+		command  any
+		response map[string]any
```
Comment (AlekSi) [inline integration/commands_administration_test.go:208]: Unless it is very hard, we should compare `bson.D`s to also check fields order

File: integration/query_projection_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		projection any
+		filter     any
+		expected   bson.D
+	}{
+		"FindProjectionInclusions": {
+			filter: bson.D{{"_id", "document-composite"}},
+			// TODO: fix projection
```
Comment (AlekSi) [inline integration/query_projection_test.go:37]: Issue URL

File: integration/query_projection_test.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			cursor, err := collection.Find(
+				ctx,
+				tc.filter,
+				options.Find().SetProjection(tc.projection),
```
Comment (AlekSi) [inline integration/query_projection_test.go:61]: That seems to be short enough to put on one line. One of the `expectred` documents above is longer

File: internal/handlers/handler_fuzz_test.go [hunk 5]
```diff
@@ -17,14 +17,55 @@ package handlers
...
+	noLogging bool
+	poolOpts  *testutil.PoolOpts
+}
+
+// setup creates shared objects for testing.
+//
+// Using shared objects helps us spot concurrency bugs.
+// If some test is failing and the log output is confusing, and you are tempted to move setup call to subtest,
+// instead run that single test with `go test -run test/name`.
+func setup(t testing.TB, opts *setupOpts) (context.Context, *pg.Handler, *pgdb.Pool) {
```
Comment (AlekSi) [inline internal/handlers/handler_fuzz_test.go:43]: I don't think that all should be in that file – that one is about fuzzing

**Review by AlekSi — COMMENTED** [review]
Nice catch with a typo in file name!

File: integration/commands_administration_test.go (hunk 2 above)
Comment (w84thesun) [reply integration/commands_administration_test.go:208]: I used `map[string]any` to skip fields like `gitVersion` or `debug` field which requires me to use `util/version` package.
If it's ok to use it then I can change `response` field to `bson.D`.

File: internal/handlers/handler_fuzz_test.go (hunk 5 above)
Comment (w84thesun) [reply internal/handlers/handler_fuzz_test.go:43]: Should I leave that code in `handler_test.go`?

File: integration/query_projection_test.go (hunk 4 above)
Comment (w84thesun) [reply integration/query_projection_test.go]: Fixed

File: integration/query_projection_test.go (hunk 3 above)
Comment (w84thesun) [reply integration/query_projection_test.go]: Fixed

File: integration/commands_administration_test.go (hunk 1 above)
Comment (w84thesun) [reply integration/commands_administration_test.go]: Fixed

**Comment by w84thesun** [comment]
> Looking into coverage report, I wonder if we deleted some tests without moving them. https://app.codecov.io/gh/FerretDB/FerretDB/compare/523/changes shows lines like that:
> 
> image
> 
> Did we forgot tests for negative results?

I've compared tests, and added one missing and two more new ones. The coverage shows those tests which now is rather disabled (stats) or not working correctly (projection).

File: integration/commands_administration_test.go (hunk 2 above)
Comment (w84thesun) [reply integration/commands_administration_test.go]: I've changed to `bson.D` for now.

File: internal/handlers/handler_fuzz_test.go (hunk 5 above)
Comment (AlekSi) [reply internal/handlers/handler_fuzz_test.go:43]: Ah, I see. Yeah, let's leave it there for now

**Review by AlekSi — APPROVED** [review]

**Pull request merged**